### PR TITLE
Limiting the number of files read at the same time

### PR DESCRIPTION
**Pull request by kiboke**
Related to https://github.com/mozilla-jetpack/jpm/issues/460

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -73,13 +77,20 @@ function zipBuffer (rootDir, options, callback) {
...
+          fs.readFile(fullPath, function (err, data) {
+            if (options.each) {
+              options.each(path.join(dir, file));
+            }
+            folders[dir].file(file, data);
+            openFiles--;
+            cb(err);
+          });
+        }
+        else
```
Comment (kumar303) [inline index.js]: this should have braces around to avoid future bugs (in case someone refactors it without looking too closely)

**Comment by kiboke** [comment]
braces added.

**Comment by kumar303** [comment]
For background/context: several `jpm` Linux users are running into an issue where jpm creates empty files for an add-on that has a massive amount of files. One fix is to increase ulimit on their system. This patch offers a workaround by buffering the amount of open files.

File: index.js [hunk 2]
```diff
@@ -73,13 +77,20 @@ function zipBuffer (rootDir, options, callback) {
...
+            if (options.each) {
+              options.each(path.join(dir, file));
+            }
+            folders[dir].file(file, data);
+            openFiles--;
+            cb(err);
+          });
+        }
+        else {
+          addItem (fullPath, cb);
```
Comment (jsantell) [inline index.js]: Remove space between function name and parens: `addItem(fullPath, cb)`

**Comment by jsantell** [comment]
For future usage, may be better to use a real queuing system (rather than calling fs.stat repeatedly until there's room).

Once the style change is fixed and commits squashed down to one commit, I'll merge this in and increment the npm package. Thank you for your contribution!

**Comment by kumar303** [comment]
@kiboke here is some info about how to squash commits: http://blog.steveklabnik.com/posts/2012-11-08-how-to-squash-commits-in-a-github-pull-request

File: index.js (hunk 2 above)
Comment (kiboke) [inline index.js]: removed.

**Comment by kiboke** [comment]
@kumar303 i think i did it, i hope it's ok.

**Comment by kumar303** [comment]
Thanks @jsantell !

**Pull request merged**